pkg/transforms/compression.go

```diff
@@ -56,18 +54,14 @@ func (compression *Compression) CompressWithGZIP(ctx interfaces.AppFunctionConte
 	}
 	var buf bytes.Buffer
 
-	if compression.gzipWriter == nil {
-		compression.gzipWriter = gzip.NewWriter(&buf)
-	} else {
-		compression.gzipWriter.Reset(&buf)
-	}
+	gzipWriter := gzip.NewWriter(&buf)
 
-	_, err = compression.gzipWriter.Write(rawData)
+	_, err = gzipWriter.Write(rawData)
 	if err != nil {
 		return false, fmt.Errorf("unable to write GZIP data in pipeline '%s': %s", ctx.PipelineId(), err.Error())
 	}
 
-	err = compression.gzipWriter.Close()
+	err = gzipWriter.Close()
 	if err != nil {
 		return false, fmt.Errorf("unable to close GZIP data in pipeline '%s': %s", ctx.PipelineId(), err.Error())
 	}
```

Comment:
if gzipWriter somehow having error on writing, should we call close() too?  i.e. make close into defer func